Allos.Amazon.Sdk/S3/Transfer/Commands/MultipartUploadCommand.cs

```diff
@@ -39,15 +37,14 @@ internal class MultipartUploadCommand : BaseCommand
         /// <summary>
         /// Initializes a new instance of the <see cref="MultipartUploadCommand"/> class.
         /// </summary>
-        /// <param name="s3Client">The s3 client.</param>
-        /// <param name="config">The config object that has the number of threads to use.</param>
+        /// <param name="asyncTransferUtility">The <see cref="IAsyncTransferUtility"/> that created this command.</param>
         /// <param name="fileTransporterRequest">The file transporter request.</param>
         /// <param name="logger">Logger</param>
         internal MultipartUploadCommand(
-            IAmazonS3 s3Client, 
-            AsyncTransferConfig config, 
+            IAsyncTransferUtility asyncTransferUtility,
             UploadRequest fileTransporterRequest,
             ILogger logger)
+            : base(asyncTransferUtility, fileTransporterRequest)
```

Comment:
Cleanup() can be made async (by making WaitForAllTasks and AbortMultipartUpload can be async too)

```
protected virtual async Task WaitForAllTasksAsync(List<Task<UploadPartResponse>> tasks)
...
   await Task.WhenAll(tasks.Cast<Task>())
            .WaitAsync(TimeSpan.FromMilliseconds(Config.MultipartUploadFinalizeTimeout));
```

and 

```
protected virtual async Task AbortMultipartUploadAsync(string uploadId)
...
   await S3Client.AbortMultipartUploadAsync(new AbortMultipartUploadRequest
```

Reply:
updated as you suggested in a more recent commit